internal/freelist/freelist.go

```diff
@@ -46,13 +46,14 @@ type Interface interface {
 	ReleasePendingPages()
 
 	// Free releases a page and its overflow for a given transaction id.
-	// If the page is already free then a panic will occur.
+	// If the page is already free or is one of the meta pages, then a panic will occur.
 	Free(txId common.Txid, p *common.Page)
 
 	// Freed returns whether a given page is in the free list.
 	Freed(pgId common.Pgid) bool
 
 	// Rollback removes the pages from a given pending tx.
+	// Should always be followed by Reload or NoSyncReload from last freelist state.
```

Comment:
Such restriction isn't friendly to users. We should consider to get rid of such restriction in following PRs. We can discuss this separately.
```suggestion
```

Reply:
applied